src/prefect/client/orchestration.py

```diff
@@ -146,6 +175,12 @@
 T = TypeVar("T")
 
 
+BaseResult.model_rebuild()
+ResultRecordMetadata.model_rebuild()
+
+State.model_rebuild()
```

Comment:
We do this in `prefect.main` too. Do we need to rebuild again here? Or is there a way that we could conditionally rebuild?

Reply:
if I don't then I get errors like this

```python
FAILED tests/client/test_prefect_client.py::test_set_then_read_flow_run_state - pydantic.errors.PydanticUndefinedAnnotation: name 'State' is not defined
```

though I agree it seems like it shouldn't be needed 🤔

Reply:
ill make a discussion to investigate this